### PR TITLE
[resotocore][feat] add the infrastructure apps cli command

### PR DESCRIPTION
**Pull request by meln1k**
# Description

Added an infrastructure apps cli command with a dummy implementation. Waiting for https://github.com/someengineering/resoto/pull/1535 to wire things together.

# To-Dos

- [ ] Add test coverage for new or updated functionality
- [x] Lint and test with `tox`
- [ ] Document new or updated functionality (someengineering/resoto.com#XXXX)

# Code of Conduct

By submitting this pull request, I agree to follow the [code of conduct](https://resoto.com/code-of-conduct).

### CODE REVIEW

File: resotocore/resotocore/cli/command.py [hunk 1]
```diff
@@ -4608,6 +4608,123 @@ async def create_certificate(
...
+    @property
+    def name(self) -> str:
+        return "apps"
+
+    def info(self) -> str:
+        return "Manage infrastructure apps."
+
+    def args_info(self) -> ArgsInfo:
+        return {
+            "search": [ArgInfo(None, True, help_text="Pattern for searching for apps.")],
```
Comment (aquamatthias) [inline resotocore/resotocore/cli/command.py:4652]: The help text is visible via tab completion on the command.
A standard terminal has 80 characters. (the help text here is already above 40 caracters + plus name + plus position of the prompt) - I would opt to shorten the help text - e..g. `<pattern>` 
Also applies to the rest of args.

File: resotocore/resotocore/cli/command.py [hunk 2]
```diff
@@ -4608,6 +4608,123 @@ async def create_certificate(
...
+    - `app update <app_name>|all`: Update an app or all apps.
+    - `apps list`: List all installed apps.
+    - `app run <app_name> [--dry-run] [--validate] [--config <config_name>]`: Run an app.
+
+
+    ## Parameters
+    - <app_name>: The identifier of the infrastructure app.
+    - [pattern]: Pattern for searching for apps. Supports glob wildcards such as * and ?.
+    - [--dry-run]: Run the app but do not make any changes.
+    - [--validate]: Validate the app but do not run it.
```
Comment (aquamatthias) [inline resotocore/resotocore/cli/command.py:4639]: Parameters and options are listed separately.

File: resotocore/resotocore/cli/command.py [hunk 3]
```diff
@@ -4608,6 +4608,123 @@ async def create_certificate(
             return CLISource.single(lambda: stream.just(self.rendered_help(ctx)))
 
 
+class InfrastructureAppsCommand(CLICommand):
+    """
+    ```shell
+    apps search [pattern] [someengineering/resoto-apps]
+    app info <app_name> [someengineering/resoto-apps]
+    app install <app_name> [someengineering/resoto-apps]
```
Comment (aquamatthias) [inline resotocore/resotocore/cli/command.py:4616]: I would opt to define the repository as parameter. `--repo foo/bla`.
This parameter defaults to `someengineering/resoto-apps`.

**Review by aquamatthias — COMMENTED** [review]
Thanks. Please find my comments.

File: resotocore/resotocore/cli/command.py (hunk 3 above)
Comment (lloesche) [reply resotocore/resotocore/cli/command.py:4616]: Good idea! Will update the design doc.

**Review by aquamatthias — DISMISSED** [review]
Thanks

**Review by aquamatthias — DISMISSED** [review]

**Review by aquamatthias — APPROVED** [review]

**Pull request merged**